ctms/app.py

```diff
@@ -179,6 +184,70 @@ def get_contacts_by_ids(
     ]
 
 
+def extractor_for_bulk_encoded_details(after: str) -> Tuple[str, datetime]:
+    str_decode = base64.urlsafe_b64decode(after)
+    result_after_list = str(str_decode.decode("utf-8")).split(",")
+    after_email_id = result_after_list[0]
+    after_start_time = dateutil.parser.parse(result_after_list[1])
+    return after_email_id, after_start_time
+
+
+def compressor_for_bulk_encoded_details(last_result: ContactSchema):
+    last_email_id = last_result.email.email_id
+    last_update_time = last_result.email.update_timestamp
+    result_after_encoded = base64.urlsafe_b64encode(
+        f"{last_email_id},{last_update_time}".encode("utf-8")
+    )
+    return result_after_encoded.decode()
+
+
+def get_bulk_contacts_by_timestamp(
+    db: Session,
+    start_time: datetime,
+    end_time: datetime,
+    limit: int = 10,
+    after: str = None,
+) -> CTMSBulkResponse:
+    """Get bulk contacts by time range."""
+    after_email_id = None
+    after_start_time = start_time
+    if after is not None:
+        after_email_id, after_start_time = extractor_for_bulk_encoded_details(
+            after=after
+        )
+
+    results = get_bulk_contacts(
+        db=db,
+        start_time=after_start_time,
+        end_time=end_time,
+        limit=limit,
+        after_email_id=after_email_id,
+    )
+    if len(results) > 0:
```

Comment:
This works, and there is a next URL until no results are returned. I'm thinking if ``len(results) < limit`` would work as well - results for the last set, but no next URL. However, the machine calling this API probably doesn't care.

Reply:
Modified the code for this